lib/led/ledcontrol.js

```diff
@@ -704,6 +711,128 @@ Controllers = {
+        this.io.pinMode(this.pins[pin], this.io.MODES.OUTPUT);
+      }, this);
+
+      this.ledMatrix = [];
+
+      this.sendCommand(this.OP.SYS_EN);
+      this.sendCommand(this.OP.LED_ON);
+      this.sendCommand(this.OP.BLINK_OFF);
+      this.sendCommand(this.OP.MASTER_MODE);
+      this.sendCommand(this.OP.INT_RC);
+      // hardcoding this for now. need to figure out a nice way of configuring this.
+      this.sendCommand(this.OP.COMMON_16NMOS);
+      this.sendCommand(this.OP.PWM_CONTROL | 0xF);
+    },
+
+    clearScreen: function() {
+      var width = 24, height = 16;
+      for (i=0; i<=(width*height/8); i++) {
+        this.ledMatrix[i] = Math.floor( Math.random() * 256 );
+      }
+      this.writeScreen();
+    },
+
+    /**
+     * Write the data in the buffers to the screen.
+     * @return {[type]} [description]
+     */
+    writeScreen: function() {
+      this.io.digitalWrite(this.pins["cs"], 0);
+      this.writeData(0x5, 3);
+
+      this.writeData(0x0, 7);
+
+      var width = 24, height = 16;
+      for (i=0; i <= (width * height / 8); i+=2) {
+        var data = this.ledMatrix[i];
+        data <<= 8;
+        data |= this.ledMatrix[i+1];
+
+        this.writeData(data, 16);
+      }
+
+      this.io.digitalWrite(this.pins["cs"], 1);
+    },
+
+    /**
+     * Send a command to the controller.
+     * @param  {Number} commandCode The command code.
+     * @return {[type]}             [description]
+     */
+    sendCommand: function(commandCode) {
+      // Set the message ID bits to command mode (0b100) and shift them into place.
+      // See datasheet, page 10.
+      var data = 0x4 << 8;
+      data |= commandCode;
+      data <<= 1;
+
+      this.io.digitalWrite(this.pins["cs"], 0);
+
+      // HT1632::writedata
+      this.io.pinMode(this.pins["data"], this.io.MODES.OUTPUT);
+      // Iterate through each bit of data and send them individually.
+      for ( var i = 12; i > 0; i-- ) {
+        this.io.digitalWrite(this.pins["write"], 0);
+        if ( data & ( 1 << (i-1) ) ) {
+          this.io.digitalWrite(this.pins["data"], 1);
+        } else {
+          this.io.digitalWrite(this.pins["data"], 0);
+        }
+        this.io.digitalWrite(this.pins["write"], 1);
+      }
+      this.io.pinMode(this.pins["data"], this.io.MODES.INPUT);
+      // end HT1632::writedata
+
+      this.io.digitalWrite(this.pins["cs"], 1);
+    },
+
+    //
+    writeData(data, numberOfBits) {
+      this.io.pinMode(this.pins["data"], this.io.MODES.OUTPUT);
+      // Iterate through each bit of data and send them individually.
+      for ( var i = numberOfBits; i > 0; i-- ) {
+        this.io.digitalWrite(this.pins["write"], 0);
+        if ( data & ( 1 << (i-1) ) ) {
+          this.io.digitalWrite(this.pins["data"], 1);
+        } else {
+          this.io.digitalWrite(this.pins["data"], 0);
+        }
+        this.io.digitalWrite(this.pins["write"], 1);
+      }
+      this.io.pinMode(this.pins["data"], this.io.MODES.INPUT);
```

Comment:
For all occurrences, just use dot access: `this.pins["write"]` => `this.pins.write`